File: Backend/TemuFansUsuarioBO/src/pe/edu/pucp/usuario/BO/ClienteBO.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package pe.edu.pucp.usuario.BO;

import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.ArrayList;
import java.util.Date;
import pe.edu.pucp.usuario.dao.ClienteDAO;
import pe.edu.pucp.usuario.daoImpl.ClienteDAOImpl;
import pe.edu.pucp.usuario.model.Cliente;
import pe.edu.pucp.usuario.model.Tipo_Doi;
import pe.edu.pucp.usuario.model.Tipo_Usuario;

/**
 *
 * @author usuario
 */
public class ClienteBO {
    private final ClienteDAO clienteDAO;
    
    public ClienteBO(){
        this.clienteDAO = new ClienteDAOImpl();
    }   
    public ArrayList<Cliente> listarTodos(){
        return clienteDAO.listarTodos();
    }

    public ArrayList<Cliente> listarTodosNombreCodigo(String nombre, String codigo){
        return clienteDAO.listarTodosNombreCodigo(nombre, codigo);
    }
    
    public ArrayList<Cliente> buscarCliente(String nombre){
        return clienteDAO.listarPorNombre(nombre);
    }

    public Integer insertarCliente(String tipo_usuario, String doi, String tipo_doi,
            String correo, String fechaRegistro, String nombre, String telefono) throws ParseException{
        
        
        Tipo_Usuario tipUs = Tipo_Usuario.valueOf(tipo_usuario);
        Tipo_Doi tipDoi = Tipo_Doi.valueOf(tipo_doi);
        
        SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd");
        Date fecha = sdf.parse(fechaRegistro);
        
        Cliente cliente = new Cliente(null, tipUs, doi, tipDoi, correo, 
                fecha, nombre, null, telefono);
        
//        Cliente cliente = new Cliente(tipo_usuario, doi, tipo_doi, correo, 
//                fechaRegistro, nombre, telefono);
        
        return this.clienteDAO.insertar(cliente);
    }
    
    public boolean existeCliente(String doi){
        return this.clienteDAO.existeCliente(doi);
    }
    
    public Cliente obtenerPorId(Integer idCliente){
        return this.clienteDAO.obtenerPorId(idCliente);
    }
}

File: Front/TemuFansUsuarioBO/ClienteBO.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using TemuFansBO;
using TemuFansBO.ServicioWS;

namespace TemuFansUsuarioBO
{
    public class ClienteBO : BaseBO
    {
        public BindingList<cliente> listarTodos()
        {
            cliente[] lista = this.ServicioWS.cliente_listarClientes();
            return new BindingList<cliente>(lista);
        }

        public BindingList<cliente> listarTodosNombreCodigo(string nombre, string codigo)
        {
            cliente[] lista = this.ServicioWS.cliente_listarClientesNombreCodigo(nombre, codigo);
            return new BindingList<cliente>(lista);
        }
        public int insertarCliente(string tipo_usuario, string doi, string tipo_doi, string correo, string fechaRegistro, string nombre, string telefono)
        {
            return this.ServicioWS.usuario_insertar_cliente(tipo_usuario, doi, tipo_doi, correo, fechaRegistro, nombre, telefono);
        }

        public Boolean existeCliente(string doi)
        {
            return this.ServicioWS.usuario_existe_cliente(doi);
        }

        public BindingList<cliente> buscarCliente(string nombre)
        {
            cliente[] lista = this.ServicioWS.cliente_buscarCliente(nombre);
            // Si lista es null, crea un array vacío
            if (lista == null)
            {
                lista = new cliente[0];
            }
            return new BindingList<cliente>(lista);
        }

        public cliente obtenerPorId(int idCliente)
        {
            return this.ServicioWS.cliente_obtenerPorId(idCliente);
        }
    }
}
